### PR TITLE
Use standard create/update/delete for supports? checks

### PR DESCRIPTION
**Pull request by agrare**
Rather than using "special" supports feature names we should prefer standard create/update/delete where possible

Depends on:
- [ ] https://github.com/ManageIQ/manageiq-providers-openstack/pull/754
- [ ] https://github.com/ManageIQ/manageiq-providers-amazon/pull/727
- [ ] https://github.com/ManageIQ/manageiq-providers-ibm_cloud/pull/303
- [ ] https://github.com/ManageIQ/manageiq-ui-classic/pull/7954
- [ ] https://github.com/ManageIQ/manageiq/pull/21550

### CODE REVIEW

**Comment by agrare** [comment]
@miq-bot cross-repo-tests /all, manageiq#21550, manageiq-ui-classic#7954, manageiq-providers-ibm_cloud#303, manageiq-providers-amazon#727, manageiq-providers-openstack#754

**Comment by Fryguy** [comment]
@agrare Is this going to collide with #1097 ?  If so, should we get that one in first, since it's probably harder to refactor than this one? cc @kbrock

**Comment by agrare** [comment]
@Fryguy real quick search doesn't look like this will conflict with that one, but no issue holding off on this until that one is merged I have a bit of work to do to get this whole PR set green anyway

**Comment by agrare** [comment]
Okay cross-repo tests are green, taking all of these out of WIP

**Comment by kbrock** [comment]
the other PR is 9 commits. So would be nice to get that in first ;)
I agree, it doesn't look like there should be much collision.

**Review by kbrock — APPROVED** [review]
looks great

File: app/controllers/api/host_aggregates_controller.rb
Comment (kbrock) [inline app/controllers/api/host_aggregates_controller.rb:8]: YAY. all three of these are nice to move over to supports

**Review by kbrock — APPROVED** [review]
looks great

**Comment by kbrock** [comment]
@agrare  My only question is on the ui classic one. I'm guessing we want to merge this and then introduce changes from https://github.com/ManageIQ/manageiq/pull/21553 ?

Or are you thinking that you want to get the subclass supports into there and then refactor that PR?

**Comment by agrare** [comment]
> Or are you thinking that you want to get the subclass supports into there and then refactor that PR?

I started out not wanting to introduce too many other changes into that PR so I worked with what I had but I'd love to get the `subclasses_supporting` PR in first then use that instead since it looks really close/already-ready

**Comment by chessbyte** [comment]
I guess the natural followup to this PR set is to change the _queue method name as well.

Something like

```
raise BadRequestError, klass.unsupported_reason(:create) unless klass.supports?(:create)
task_id = klass.create_queue(session[:userid], ext_management_system, data)
```

instead of

```
raise BadRequestError, klass.unsupported_reason(:create) unless klass.supports?(:create)
task_id = klass.create_key_pair_queue(session[:userid], ext_management_system, data)
```

**Pull request merged**
